Add spatial visualisations for decision matrices

Rosejoycrocker opened the pull request:
Adds method to viz.map for plotting decision matrices and aggregate scores on a set of side-by-side maps.

Rosejoycrocker commented:
Here is the plot from running the example code in analysis.md:

![criteria_plots_hs](https://github.com/user-attachments/assets/231014ed-7e61-4676-a96c-8a40d025b40b)

(any advice on what the grey boxes are appreciated, I'll try to remove)

Zapiano commented:
@Rosejoycrocker this PR has a conflict and the tests are not passing. About the tests, this could be due to a problem that was fixed yesterday, not sure.

Rosejoycrocker commented:
> @Rosejoycrocker this PR has a conflict and the tests are not passing. About the tests, this could be due to a problem that was fixed yesterday, not sure.

I'm looking into it now

Rosejoycrocker commented:
Tests are passing and conflicts reolved now @Zapiano :)

Rosejoycrocker commented on src/decision/Criteria/DecisionPreferences.jl:
```diff
@@ -131,6 +131,30 @@ Index of locations ordered by their rank
 function rank_by_index(
     dp::T, dm::YAXArray, method::Union{Function,DataType}
 )::Vector{Int64} where {T<:DecisionPreference}
+    res = get_criteria_aggregate(dp, dm, method)
```
Its a result structure that the MCDA package uses, so it contains more than just the aggregate score. Maybe change it to `decision_results` or something?

Rosejoycrocker commented on src/decision/Criteria/DecisionPreferences.jl:
```diff
@@ -131,6 +131,30 @@ Index of locations ordered by their rank
+# Returns
+Returns raw aggreagted score for a set of locations
+"""
+function get_criteria_aggregate(
```
Thanks for spotting, I'll fix the signature and change the name to `criteria_aggregated_scores`

Rosejoycrocker commented on ext/AvizExt/viz/spatial.jl:
```diff
@@ -251,6 +256,75 @@ function ADRIA.viz.map!(
         axis_opts
     )
 end
+function ADRIA.viz.map(
```
Sounds good :)

ConnectedSystems replied:
I prefer the current approach but I'm not 100% up on the context here, just latching onto the suggestion to create a separate map plotting function. Julia is not strictly an object oriented language and isn't intended to be used like Python or Java.

The advantage of having multiple dispatch (or something like it, as in R) is to prevent everyone from having to remember a lot of function names. Is it `map` or `vizualize_selection_criteria_as_map` or `display_map` or `plot_selection_criteria_map` or...

As a user I don't want to have to remember endless possible function names for specific use cases. I just want a map.

Instead, this behaviour of:

1. You have a vector of results -> `map()` produces a single map.
2. You have a matrix of results -> `map()` produces a series of maps.

is sensible to me.

But again, I don't have full context.

https://www.juliaopt.org/meetings/santiago2019/slides/stefan_karpinski.pdf

Zapiano replied:
I understand and I agree with what you said. 

My problem with how this function is defined right now is that it doesn't plot selection criteria scores in particular. It plots multiple maps with whatever you pass as an argument. But still the names of variables and the plots default titles are related to this specific kind of plot (multiple maps with selection criteria scores). It's like having a function called `sum` and the arguments are called `prime_number_a` and `prime_number_b`. If it sums more than just prime numbers, they should be called `number_a` and `number_b` or whatever.

About the need for a specific function called `selection_criteria_map`, maybe we don't need one, I agree. depending on how this `map` function with multiple maps is defined we can just use this one and that's fine. My comment was more about making this `map` general than about the need of a specific function with a specific name for this kind of plot.

ConnectedSystems replied:
Incidentally @Rosejoycrocker the variable names/titles displayed in the figure should be run through `human_readable()`

Rosejoycrocker commented:
![criteria_plots](https://github.com/user-attachments/assets/86c408fd-427e-4430-b28f-bb0fb017ecb6)
Most recent version of the plotting output (grid has been removed)

Rosejoycrocker commented:
I think I've addressed your comments @Zapiano , let me know if you have others

Rosejoycrocker commented:
> I think I've addressed your comments @Zapiano , let me know if you have others

Could you give this a look over too? Thanks @Zapiano :)

Rosejoycrocker commented:
Could you give this another look @Zapiano? Thanks!

Zapiano commented:
> Could you give this another look @Zapiano? Thanks!

Hey @Rosejoycrocker , it looks like some checks haven't passed because some lines were added but not tested.

Rosejoycrocker commented:
> > Could you give this another look @Zapiano? Thanks!
> 
> Hey @Rosejoycrocker , it looks like some checks haven't passed because some lines were added but not tested.

Seems like I added tests to a file which is no longer included in `runtests.jl` (`site_selection.jl`). I just added them to `mcda.jl` instead and they're passing locally.

Zapiano commented:
> > > Could you give this another look @Zapiano? Thanks!
> > 
> > 
> > Hey @Rosejoycrocker , it looks like some checks haven't passed because some lines were added but not tested.
> 
> Seems like I added tests to a file which is no longer included in `runtests.jl` (`site_selection.jl`). I just added them to `mcda.jl` instead and they're passing locally.

It looks like some tests are not passing :/

Rosejoycrocker commented:
> > > > Could you give this another look @Zapiano? Thanks!
> > > 
> > > 
> > > Hey @Rosejoycrocker , it looks like some checks haven't passed because some lines were added but not tested.
> > 
> > 
> > Seems like I added tests to a file which is no longer included in `runtests.jl` (`site_selection.jl`). I just added them to `mcda.jl` instead and they're passing locally.
> 
> It looks like some tests are not passing :/

Tests are passing locally now for me but don't seem to be running on here anymore, maybe you could run them locally to check?
